grpc/protobuf/src/main/scala/sttp/tapir/grpc/protobuf/EndpointToProtobufMessage.scala

```diff
@@ -0,0 +1,88 @@
+package sttp.tapir.grpc.protobuf
+
+import sttp.tapir.SchemaType.{SProduct, SString}
+import sttp.tapir._
+import sttp.tapir.grpc.protobuf.model._
+
+class EndpointToProtobufMessage {
+  def apply(es: List[AnyEndpoint]): List[ProtobufMessage] = {
+    es.flatMap(forEndpoint)
+  }
+
+  private def distinctBy[T, U](elements: List[T])(selector: T => U): List[T] = elements
+    .groupBy(selector)
+    .flatMap {
+      case (_, Seq(el)) => Seq(el)
+      case _            => Seq.empty[T]
+    }
+    .toList
+
+  private def forEndpoint(e: AnyEndpoint): List[ProtobufMessage] =
+    distinctBy(forInput(e.input) ++ forOutput(e.output))(_.name)
+
+  private def forInput(input: EndpointInput[_]): List[ProtobufMessage] = {
+    input match {
+      case EndpointInput.FixedMethod(_, _, _)     => List.empty
+      case EndpointInput.FixedPath(_, _, _)       => List.empty
+      case EndpointInput.PathCapture(_, codec, _) => ???
```

Comment:
I guess these need filling in - mostly with `Nil`? or do you want to report "unsupported" errors?